src/proposer/proposer_logic.cpp

```diff
@@ -44,32 +44,37 @@ class LogicImpl final : public Logic {
   boost::optional<proposer::EligibleCoin> TryPropose(const staking::CoinSet &eligible_coins) override {
     AssertLockHeld(m_active_chain->GetLock());
 
-    const CBlockIndex *current_tip =
-        m_active_chain->GetTip();
+    const CBlockIndex *current_tip = m_active_chain->GetTip();
     if (!current_tip) {
       return boost::none;
     }
-    const blockchain::Height current_height =
-        m_active_chain->GetHeight();
-    const blockchain::Height target_height =
-        current_height + 1;
+    const blockchain::Height current_height = m_active_chain->GetHeight();
+    const blockchain::Height target_height = current_height + 1;
+
+    int64_t best_time = std::max(m_active_chain->GetTip()->GetMedianTimePast() + 1, m_network->GetTime());
```

Comment:
Unused variable.

Reply:
This is a clear mistake, due to some refactoring. The var should be used one line below.